### PR TITLE
Add switch for client UFS fallback

### PR DESCRIPTION
**Pull request by dbw9580**
### What changes are proposed in this pull request?

Add a switch for client UFS fallback.

### Why are the changes needed?

Allow a user to optionally enable or disable this feature. UFS fallback may be undesirable in cases where the clients run in an untrusted environment, so the UFS credentials cannot be exposed to the client.

### Does this PR introduce any user facing changes?

Added a new config property `alluxio.dora.client.ufs.fallback.enabled` which defaults to `false`.

### CODE REVIEW

File: dora/shell/src/main/java/alluxio/cli/fs/command/MountCommand.java [hunk 1]
```diff
@@ -77,7 +77,10 @@ public MountCommand(FileSystemContext fsContext) {
     AlluxioProperties properties = fsContext.getClusterConf().copyProperties();
     properties.set(PropertyKey.DORA_CLIENT_READ_LOCATION_POLICY_ENABLED, false);
     AlluxioConfiguration config = new InstancedConfiguration(properties);
-    mFileSystem = FileSystem.Factory.create(fsContext, FileSystemOptions.create(config));
+    mFileSystem = FileSystem.Factory.create(fsContext,
```
Comment (yyongycy) [inline dora/shell/src/main/java/alluxio/cli/fs/command/MountCommand.java:80]: do we still support this command?

**Review by yyongycy — APPROVED** [review]
lgtm

File: dora/shell/src/main/java/alluxio/cli/fs/command/MountCommand.java (hunk 1 above)
Comment (dbw9580) [reply dora/shell/src/main/java/alluxio/cli/fs/command/MountCommand.java:80]: Not currently.

File: dora/core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java [hunk 2]
```diff
@@ -332,6 +353,9 @@ public void rename(AlluxioURI src, AlluxioURI dst, RenamePOptions options)
 
       mDoraClient.rename(srcUfsFullPath.toString(), dstUfsFullPath.toString(), mergedOptions);
     } catch (RuntimeException ex) {
+      if (!mUfsFallbackEnabled) {
```
Comment (huanghua78) [inline dora/core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java:356]: I think this is the "switch" to turn off UFS fallback.

I don't understand why the rest of the patch is needed.
The reason for the other changes are not stated in the PR description, or not related to this PR?

Comment (dbw9580) [reply dora/core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java:356]: They are needed for the condition checks here https://github.com/Alluxio/alluxio/pull/17622/files#diff-e3c565cd9781b8c3cd10e0e77d0b00ff48ce3a0efc9a9daece52938a655b8a40

Comment (dbw9580) [reply dora/core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java:356]: I needed a `isUfsFallbackEnabled` check

**Comment by dbw9580** [comment]
alluxio-bot, merge this please

**Pull request merged**